### PR TITLE
Submit comment tests

### PR DESCRIPTION
**Pull request by vrajmohan**
@cmc333333, @jmcarp : Please review. I am writing more tests.

### CODE REVIEW

File: regulations/tests/tasks_tests.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                                           retry, save_failed_submission):
+        html_to_pdf.return_value.__enter__ = mock.Mock(
+            return_value=self.file_handle)
+
+        post_submission.side_effect = [RequestException]
+
+        retry.return_value = MaxRetriesExceededError()
+
+        submit_comment(self.submission)
+        save_failed_submission.assert_called_with(json.dumps(self.submission))
```
Comment (jmcarp) [inline regulations/tests/tasks_tests.py]: Minor, but what do you think about verifying that the failed submission actually made it to the database? I have a feeling that @cmc333333 might point out that that would make this more of an integration tests, since it would be testing multiple units, but I don't necessarily see that as a bad thing.

This has been a conversation between me and imaginary @cmc333333.

File: regulations/tests/tasks_tests.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    def setUp(self):
+        self.file_handle = six.BytesIO(b"some-byte-content")
+        self.submission = {'assembled_comment': [
+            {"id": "A1", "comment": "A simple comment", "files": []},
+            {"id": "A5", "comment": "Another comment", "files": []}
+        ]}
+
+    def test_submit_comment(self, html_to_pdf, post_submission, retry,
+                            save_failed_submission):
+        html_to_pdf.return_value.__enter__ = mock.Mock(
```
Comment (jmcarp) [inline regulations/tests/tasks_tests.py]: Minor, but since `html_to_pdf` is mocked, all of its attributes are also recursively mocked. We can shorten this to `html_to_pdf.return_value.__enter__.return_value = self.file_handle`.

File: regulations/tests/tasks_tests.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        result = submit_comment(self.submission)
+
+        self.assertEqual(result, expected_result)
+
+    def test_failed_submit_raises_retry(self, html_to_pdf, post_submission,
+                                        retry, save_failed_submission):
+        html_to_pdf.return_value.__enter__ = mock.Mock(
+            return_value=self.file_handle)
+
+        post_submission.side_effect = [RequestException]
```
Comment (jmcarp) [inline regulations/tests/tasks_tests.py]: Curious: why is this a list?

**Comment by jmcarp** [comment]
This looks good to me. I posted a few comments, all minor, but nothing that should hold up merge. Also, now that `submit_comment` has been broken up into smaller functions, we'll want to write tests for those functions too--which can also happen in a separate pull request.

@vrajmohan: I'm happy to merge as-is, or after revisions, if you agree with my suggestions.

File: regulations/tests/tasks_tests.py (hunk 1 above)
Comment (vrajmohan) [inline regulations/tests/tasks_tests.py]: Music to my ears. If your remember, I was part of that conversation too and I much prefer real tests  to mocking. I must have been swayed to the dark side.

File: regulations/tests/tasks_tests.py (hunk 3 above)
Comment (vrajmohan) [inline regulations/tests/tasks_tests.py]: At one time, I was planning to make multiple calls, thinking that I would somehow use Celery's retry mechanism to fail a few tries. I didn't remove it when I went another way.

**Comment by vrajmohan** [comment]
Thanks for your review, I'll address them.

File: regulations/tests/tasks_tests.py (hunk 1 above)
Comment (cmc333333) [inline regulations/tests/tasks_tests.py]: I worry that you all think I'm much more absolutist than I think I am! If the purpose of this function is to save something to the database, the simplest validation would be to query the database :)

**Comment by jmcarp** [comment]
Thanks for updating!
